### PR TITLE
Add backlinks to the change course flow

### PR DESCRIPTION
**Pull request by JR-G**
## Context

Held off adding backlinks to this feature until the entire flow was in place. It's now time to add them.

## Changes proposed in this pull request

- add `CoursePathHelper`
- add a backlink to each stage of the flow

## Guidance to review

Found an existing bug if you follow this scenario:
- Change course
- Change location
- On the check page go back one step (to location)
- On the location page go back another step (to course)

This causes an error.

## Link to Trello card

https://trello.com/c/ivqYMCU4

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: spec/helpers/course_path_helper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      it 'returns the application choice path' do
+        expect(helper.course_path_for(application_choice, 'select_option'))
+          .to eq(provider_interface_application_choice_path(application_choice, {}))
+      end
+    end
+
+    context 'when any other step' do
+      it 'returns the step edit path' do
+        expect(helper.course_path_for(application_choice, :other_step))
+          .to eq([:edit, :provider_interface, application_choice, :course, :other_step, params])
```
Comment (SamJamCul) [inline spec/helpers/course_path_helper_spec.rb:17]: I'm surprised that you could put `params` in the expectation! Could you make it more explicit that the `params` being passed in are the ones expected to come out?

File: spec/system/provider_interface/provider_changes_a_course_before_point_of_offer_spec.rb [hunk 2]
```diff
@@ -215,6 +219,12 @@ def and_i_select_a_course_with_one_study_mode_and_one_location
     choose @one_mode_and_location_course.name_and_code
   end
 
+  def when_i_click_back_i_go_to_the_course_page
+    click_on 'Back'
+    expect(page).to have_content "Update course - #{application_form.full_name}"
+    expect(page).to have_content 'Course'
+  end
```
Comment (SamJamCul) [inline spec/system/provider_interface/provider_changes_a_course_before_point_of_offer_spec.rb:226]: Could split this into two separate methods, one's an action and the other's an expectation.

Comment (JR-G) [reply spec/system/provider_interface/provider_changes_a_course_before_point_of_offer_spec.rb:226]: Yeah fair point – I mean, this method already exists in `then_i_see_a_list_of_courses_to_select_from`. I'm not 100% sure on how i feel about the testing approach here – feel it needs to be a bit more in depth in general but also don't want to inflate the spec too much 🤔

Comment (SamJamCul) [reply spec/system/provider_interface/provider_changes_a_course_before_point_of_offer_spec.rb:226]: Yeah the back link is mainly based on that `wizard.previous_step` which is already tested somewhere else. I think it's probably the right amount of depth to check that it works as implemented, and not every single page.

**Review by cwrw — APPROVED** [review]
👍 spotted the location bug you mention (unrelated to this) can tackle in a different PR

**Pull request merged**